{
  "modified": "2025-06-03T20:53:33Z",
  "published": "2025-05-15T13:29:20Z",
  "schema_version": "1.5.0",
  "id": "CVE-2025-4516",
  "aliases": [
    "PSF-2025-4"
  ],
  "details": "There is an issue in CPython when using `bytes.decode(\"unicode_escape\", error=\"ignore|replace\")`. If you are not using the \"unicode_escape\" encoding or an error handler your usage is not affected. To work-around this issue you may stop using the error= handler and instead wrap the bytes.decode() call in a try-except catching the DecodeError.",
  "affected": [
    {
      "package": {
        "ecosystem": "DHI",
        "name": "python",
        "purl": "pkg:dhi/python"
      },
      "ranges": [
        {
          "type": "SEMVER",
          "events": [
            {
              "introduced": "0"
            },
            {
              "fixed": "3.9.23"
            },
            {
              "introduced": "3.10.0-alpha1"
            },
            {
              "fixed": "3.10.18"
            },
            {
              "introduced": "3.11.0-alpha1"
            },
            {
              "fixed": "3.11.13"
            },
            {
              "introduced": "3.12.0-alpha1"
            },
            {
              "fixed": "3.12.11"
            },
            {
              "introduced": "3.13.0-alpha1"
            },
            {
              "fixed": "3.13.4"
            },
            {
              "introduced": "3.14.0-alpha1"
            },
            {
              "fixed": "3.14.0-beta2"
            }
          ],
          "repo": "https://github.com/python/cpython"
        }
      ]
    }
  ],
  "references": [
    {
      "type": "REPORT",
      "url": "https://github.com/python/cpython/issues/133767"
    },
    {
      "type": "WEB",
      "url": "https://github.com/python/cpython/pull/129648"
    },
    {
      "type": "ADVISORY",
      "url": "https://mail.python.org/archives/list/security-announce@python.org/thread/L75IPBBTSCYEF56I2M4KIW353BB3AY74/"
    },
    {
      "type": "FIX",
      "url": "https://github.com/python/cpython/commit/69b4387f78f413e8c47572a85b3478c47eba8142"
    },
    {
      "type": "FIX",
      "url": "https://github.com/python/cpython/commit/9f69a58623bd01349a18ba0c7a9cb1dad6a51e8e"
    },
    {
      "type": "FIX",
      "url": "https://github.com/python/cpython/commit/4398b788ffc1f954a2c552da285477d42a571292"
    },
    {
      "type": "FIX",
      "url": "https://github.com/python/cpython/commit/6279eb8c076d89d3739a6edb393e43c7929b429d"
    },
    {
      "type": "FIX",
      "url": "https://github.com/python/cpython/commit/73b3040f592436385007918887b7e2132aa8431f"
    },
    {
      "type": "FIX",
      "url": "https://github.com/python/cpython/commit/8d35fd1b34935221aff23a1ab69a429dd156be77"
    },
    {
      "type": "FIX",
      "url": "https://github.com/python/cpython/commit/ab9893c40609935e0d40a6d2a7307ea51aec598b"
    }
  ]
}